test/ValidateOAuthVersionMessageHandler.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace OAuth.Net.Tests
+{
+    public class ValidateOAuthVersionMessageHandler
+    {
+        [Fact]
+        public async Task ValidateDefaultOAuthVersionHandler()
+        {
+            using (var handler = new OAuthMessageHandler("apiKey", "secret", "authToken", "authTokenSecret"))
+            {
+                var oauthVersion = await GetOAuthVersionFromHandlerAsync(handler);
+
+                Assert.Equal("1.0", oauthVersion);
+            }
+        }
+
+        [Fact]
+        public async Task ValidateOAuthVersionOmitHandler()
+        {
+            using (var handler = new OAuthMessageHandler("apiKey", "secret", "authToken", "authTokenSecret", OAuthVersion.Omit))
+            {
+                var oauthVersion = await GetOAuthVersionFromHandlerAsync(handler);
+
+                Assert.Null(oauthVersion);
+            }
+        }
+
+        [Fact]
+        public async Task ValidateOAuthVersionOneZeroHandler()
+        {
+            using (var handler = new OAuthMessageHandler("apiKey", "secret", "authToken", "authTokenSecret", OAuthVersion.OneZero))
+            {
+                var oauthVersion = await GetOAuthVersionFromHandlerAsync(handler);
+
+                Assert.Equal("1.0", oauthVersion);
+            }
+        }
+
+        [Fact]
+        public async Task ValidateOAuthVersionOneZeroAHandler()
+        {
+            using (var handler = new OAuthMessageHandler("apiKey", "secret", "authToken", "authTokenSecret", OAuthVersion.OneZeroA))
+            {
+                var oauthVersion = await GetOAuthVersionFromHandlerAsync(handler);
+
+                Assert.Equal("1.0a", oauthVersion);
+            }
+        }
+
+        private async Task<string> GetOAuthVersionFromHandlerAsync(OAuthMessageHandler handler)
+        {
+            var testHandler = new TestHttpMessageHandler();
+            handler.InnerHandler = testHandler;
+
+            using (var httpClient = new HttpClient(handler, disposeHandler: false))
+            {
+                await httpClient.GetAsync("http://example.com");
```

Comment:
Is this ok to ping from tests?

Reply:
I believe it is, it can at least be used for examples without asking for permission (open http://example.com in a browser). I'm pretty sure the test is not actually doing any requests though. 

If I understand things correctly the `HttpMessageHandler` is responsible for making the request, and since the `OAuthMessageHandler` is a `DelegatingHandler` the responsibility of the request is delegated to the `InnerHandler`. I set the `InnerHandler` to an instance of `TestHttpMessageHandler`, which never does a real request but instead just returns 200 OK.

When running the tests with Fiddler active, I did not see any requests. I also tried running the tests with network disabled and they still passed. This seems to verify my beliefs.

Reply:
If we don't even end up hitting the network I don't see any issues with using that URL.